test/integration/targets/nxos_acl_interfaces/tests/cli/merged.yml

```diff
@@ -0,0 +1,74 @@
+---
+- debug:
+    msg: "Start nxos_acl_interfaces merged integration tests connection = {{ansible_connection}}"
+
+- include_tasks: populate_acl.yaml
+
+- block:
+    - name: Gather acl interfaces facts
+      nxos_facts:
+        gather_subset:
+          - "!all"
+          - "!min"
+        gather_network_resources: acl_interfaces
+
+    - name: Merged
+      nxos_acl_interfaces: &merged
+        config:
+          - name: Ethernet1/2
+            access_groups:
+              - afi: ipv6
+                acls:
+                  - name: ACL1v6
+                    direction: in
+
+          - name: Eth1/5
+            access_groups:
+              - afi: ipv4
+                acls:
+                  - name: PortACL
+                    direction: in
+                    port: True
+
+                  - name: ACL1v4
+                    direction: out
+
+              - afi: ipv6
+                acls:
+                  - name: ACL1v6
+                    direction: in
+        state: merged
+      register: result
+
+    - assert:
+        that:
+          - "result.changed == True"
+          - "'interface Ethernet1/2' in result.commands"
+          - "'ipv6 traffic-filter ACL1v6 in' in result.commands"
+          - "'interface Ethernet1/5' in result.commands"
+          - "'ip port access-group PortACL in' in result.commands"
+          - "'ip access-group ACL1v4 out' in result.commands"
+          - "'ipv6 traffic-filter ACL1v6 in' in result.commands"
+          - "result.commands | length == 6 "
+
+    - name: Gather acl interfaces facts
+      nxos_facts:
+        gather_subset:
+          - "!all"
+          - "!min"
+        gather_network_resources: acl_interfaces
+
+    - assert:
+        that:
+          - "ansible_facts.network_resources.acl_interfaces == result.after"
```

Comment:
I'm not sure I get this assertion. Please correct me if I'm wrong. The same code that is use to parse facts is also used to compute result.after. This assertion would always be true, even if the facts parsed is wrong. IMO, we should always hardcode the SoT for assertions in vars/main.yml.

Reply:
Yeah, thats right. It should have been compared with the SoT.
This check was actually redundant, as I am comparing the commands executed earlier. Removed it.